docs/source/extend_kedro/plugins.md

```diff
@@ -69,6 +69,8 @@ setup(
 )
 ```
 
+After that you can use this starter with `kedro new --starter=test_plugin_starter --checkout=main`.
+
```

Comment:
````suggestion
After that you can use this starter with `kedro new --starter=test_plugin_starter`.

```{note}
If your starter lives on a git repository, by default Kedro attempts to use a tag or branch labelled with your version of Kedro, e.g. `0.18.2.`. This means that you can host different versions of your starter template on the same repository, and the correct one will automatically be used. If you do not wish to follow this structure, you should override it with the `checkout` flag, e.g. `kedro new --starter=test_plugin_starter --checkout=main`.
```
````
@noklam I think this is clearer?

I've actually always thought (and I think @Galileo-Galilei agrees) that we should change this behaviour:
* by default don't try and checkout the kedro version (cookiecutter will by default look for the right `main`/`master` branch if you don't specify `checkout`)
* and/or enable people to specify `checkout` in their `KedroStarterSpec`
... since in reality no one maintains a starters repo which they keep up to date with every single version of the kedro template.

But that's a discussion for another time.